eng/versioning.targets

```diff
@@ -157,9 +157,10 @@
   </Target>
 
   <!-- Removes assembly level attributes for Browser. -->
-  <Target Name="RemoveSupportedPlatformAssemblyAttributeForBrowser" AfterTargets="GetAssemblyAttributes" Condition="'$(TargetFrameworkSuffix)' == 'Browser'">
+  <Target Name="RemoveSupportedPlatformAssemblyAttributeForBrowser" AfterTargets="GetAssemblyAttributes" Condition="'$(TargetFrameworkSuffix)' == 'Browser' or '$(IsTestProject)' == 'true'">
```

Comment:
why can't the attributes stay on test projects?

Reply:
`artifacts\obj\Microsoft.Extensions.Logging.Tests\net6.0-windows-Debug\Microsoft.Extensions.Logging.Tests.AssemblyInfo.cs(23,38): error CS0433: (NETCORE_ENGINEERING_TELEMETRY=Build) The type 'TargetPlatformAttribute' exists in both 'Microsoft.Extensions.Logging.EventLog, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' and 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'`

There is build error on test projects, i believe we don't want/need to add attributes to the test build, but we will need to keep the rule for that. Anyway i was not able to figure out any other way to fix this error, please suggest

Reply:
Same as [below](https://github.com/dotnet/runtime/pull/45504/files#r535730162).